pkg/agent/proxy/endpoints.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proxy
+
+import (
+	"fmt"
+	"net"
+	"reflect"
+	"sync"
+
+	corev1 "k8s.io/api/core/v1"
+	apimachinerytypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/proxy/types"
+	k8sproxy "github.com/vmware-tanzu/antrea/third_party/proxy"
+)
+
+type endpointsChange struct {
```

Comment:
I think you should add more comments to this file for structs and funcs. Not easy to understand the code without enough comments.

Reply:
Updated.